### PR TITLE
Improves restraints sprites, and adds noose and cable cuff colors

### PR DESCRIPTION
**Pull request by alsoandanswer**
Cable cuffs and nooses now use the same color as the cable coil used to make it

You can also cut cable cuffs to get back the materials.

Improves sprites for cable cuffs, handcuffs and zipties, as well as tweaking cable cuff inhands

firefox_fQiYbMkDrE
firefox_bqUdrFyFae

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items/weapons/handcuffs.dm [hunk 1]
```diff
@@ -160,9 +189,16 @@
 			to_chat(user, "<span class='notice'>You wrap the cable restraint around the top of the rod.</span>")
 			qdel(src)
 			update_icon(user)
+	else if(I.iswirecutter())
+		user.visible_message("[user] cuts the [src].", "<span class='notice'>You cut the [src].</span>")
```
Comment (Geevies) [inline code/game/objects/items/weapons/handcuffs.dm:193]:
```suggestion
		user.visible_message("<b>\The [user]</b> cuts \the [src].", SPAN_NOTICE("You cut \The [src]."))
```

**Comment by alsoandanswer** [comment]
> looks good, i assume you tested this

yeah i did

hot pink nooses are the new flex

File: code/game/objects/items/weapons/handcuffs.dm [hunk 2]
```diff
@@ -120,35 +120,64 @@
...
+		"Cyan" = COLOR_CYAN,
+		"Red" = COLOR_RED,
+		"White" = COLOR_WHITE
+	)
+
+/obj/item/handcuffs/cable/Initialize()
+	. = ..()
+	update_icon()
+
+obj/item/handcuffs/cable/update_icon()
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/handcuffs.dm:149]:
```suggestion
/obj/item/handcuffs/cable/update_icon()
```

File: code/game/objects/items/weapons/handcuffs.dm [hunk 3]
```diff
@@ -120,35 +120,64 @@
...
+	var/list/possible_cablecuff_colours = list(
+		"Yellow" = COLOR_YELLOW,
+		"Green" = COLOR_LIME,
+		"Pink" = COLOR_PINK,
+		"Blue" = COLOR_BLUE,
+		"Orange" = COLOR_ORANGE,
+		"Cyan" = COLOR_CYAN,
+		"Red" = COLOR_RED,
+		"White" = COLOR_WHITE
+	)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/handcuffs.dm:143]: This list should really be static at this point

File: code/game/objects/items/weapons/handcuffs.dm [hunk 4]
```diff
@@ -120,35 +120,64 @@
...
+	update_icon()
+
+/obj/item/handcuffs/cable/update_icon()
+	if(build_from_parts) //random colors!
+		if(!our_color)
+			our_color = pick(possible_cablecuff_colours)
+		var/color_hex = possible_cablecuff_colours[our_color]
+		color = color_hex
+		item_state = "coil-[our_color]"  // hardcoded. sucks, but inhands are hard and I can't be bothered.
+		add_overlay(overlay_image(icon, "[initial(icon_state)]_end", flags=RESET_COLOR))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/handcuffs.dm:156]: This is starting to get copypasted a bit too much for my liking

File: code/game/objects/items/weapons/handcuffs.dm [hunk 5]
```diff
@@ -160,9 +189,16 @@
 			to_chat(user, "<span class='notice'>You wrap the cable restraint around the top of the rod.</span>")
 			qdel(src)
 			update_icon(user)
+	else if(I.iswirecutter())
+		user.visible_message("<b>\The [user]</b> cuts \the [src].", SPAN_NOTICE("You cut \The [src]."))
+		var/obj/item/stack/cable_coil/C = new(get_turf(src))
+		C.our_color = our_color
+		C.amount = 15
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/handcuffs.dm:196]: Handle it in initialize

File: code/game/objects/items/weapons/handcuffs.dm (hunk 4 above)
Comment (alsoandanswer) [reply code/game/objects/items/weapons/handcuffs.dm:148]: I feel like it's out of my skill bracket to streamline this any further

File: code/modules/power/cable.dm [hunk 6]
```diff
@@ -21,6 +21,17 @@ If d1 = 0 and d2 = dir, it's a O-X cable, getting from the center of the tile to
...
+var/static/list/possible_cable_coil_colours = list(
+		"Yellow" = COLOR_YELLOW,
+		"Green" = COLOR_LIME,
+		"Pink" = COLOR_PINK,
+		"Blue" = COLOR_BLUE,
+		"Orange" = COLOR_ORANGE,
+		"Cyan" = COLOR_CYAN,
+		"Red" = COLOR_RED,
+		"White" = COLOR_WHITE
+	)
```
Comment (NonQueueingMatt) [inline code/modules/power/cable.dm:33]: Not like this, just a static list on the item

**Pull request merged**